src/provider/provider_os_memory.c

```diff
@@ -598,6 +676,110 @@ static umf_result_t os_allocation_merge(void *provider, void *lowPtr,
     return UMF_RESULT_SUCCESS;
 }
 
+typedef struct os_ipc_data_t {
+    int pid;
+    int fd;
+    size_t fd_offset;
+    size_t size;
+} os_ipc_data_t;
+
+static umf_result_t os_get_ipc_handle_size(void *provider, size_t *size) {
+    if (provider == NULL || size == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    *size = sizeof(os_ipc_data_t);
+    return UMF_RESULT_SUCCESS;
+}
+
+static umf_result_t os_get_ipc_handle(void *provider, const void *ptr,
+                                      size_t size, void *providerIpcData) {
+    if (provider == NULL || providerIpcData == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    os_memory_provider_t *os_provider = (os_memory_provider_t *)provider;
+
+    uintptr_t rkey;
+    void *rvalue;
+    int ret = critnib_find(os_provider->ptr_off, (uintptr_t)ptr, FIND_LE, &rkey,
```

Comment:
Following our offline discussion, we need to investigate why we can get a `ptr` that does not exactly match a key from the `ptr_off` map. Keys stored in the `ptr_off` map are the addresses pointed to at the beginning of the coarse grain allocations (returned by `os_alloc`).
My guess is that caused by split/merge functionality. If so, it is incorrect fix, and we should maintain the state fo this `ptr_off` map in split/merge functions.